app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt

```diff
@@ -0,0 +1,178 @@
+    }
+
+    public override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_talk_topic)
+        title = ""
+        linkHandler = TalkLinkHandler(this)
+
+        if (intent.hasExtra(EXTRA_LANGUAGE)) {
+            wikiSite = WikiSite.forLanguageCode(intent.getStringExtra(EXTRA_LANGUAGE).orEmpty())
+        }
+        userName = intent.getStringExtra(EXTRA_USER_NAME).orEmpty()
+        topicId = intent.extras?.getInt(EXTRA_TOPIC, 0)!!
+
+        talk_recycler_view.layoutManager = LinearLayoutManager(this)
+        talk_recycler_view.addItemDecoration(DrawableItemDecoration(this, R.attr.list_separator_drawable, drawStart = false, drawEnd = false))
+        talk_recycler_view.adapter = TalkReplyItemAdapter()
+
+        talk_reply_button.setOnClickListener {
+            // TODO
+        }
+
+        talk_refresh_view.setOnRefreshListener {
+            loadTopic()
+        }
+
+        talk_reply_button.visibility = View.GONE
+        loadTopic()
+    }
+
+    public override fun onDestroy() {
+        disposables.clear()
+        super.onDestroy()
+    }
+
+    private fun loadTopic() {
+        disposables.clear()
+        talk_progress_bar.visibility = View.VISIBLE
+        talk_error_view.visibility = View.GONE
+
+        disposables.add(ServiceFactory.getRest(wikiSite).getTalkPage(userName)
+                .subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .subscribe({ response ->
+                    topic = response.topics?.find { t -> t.id == topicId }
+                    updateOnSuccess()
+                }, { t ->
+                    L.e(t)
+                    updateOnError(t)
+                }))
+    }
+
+    private fun updateOnSuccess() {
+        talk_progress_bar.visibility = View.GONE
+        talk_error_view.visibility = View.GONE
+        talk_reply_button.visibility = View.VISIBLE
+        talk_refresh_view.isRefreshing = false
+
+        val titleStr = StringUtil.fromHtml(topic?.html).toString().trim()
+        title = if (titleStr.isNotEmpty()) titleStr else getString(R.string.talk_no_subject)
+        talk_recycler_view.adapter?.notifyDataSetChanged()
+    }
+
+    private fun updateOnError(t: Throwable) {
+        talk_progress_bar.visibility = View.GONE
+        talk_refresh_view.isRefreshing = false
+        talk_reply_button.visibility = View.GONE
+        talk_error_view.visibility = View.VISIBLE
+        talk_error_view.setError(t)
+    }
+
+    private fun showLinkPreviewOrNavigate(title: PageTitle) {
+        if (title.namespace() == Namespace.USER_TALK) {
+            startActivity(newIntent(this, title.wikiSite.languageCode(), title.text))
+        } else {
+            bottomSheetPresenter.show(supportFragmentManager,
+                    LinkPreviewDialog.newInstance(HistoryEntry(title, HistoryEntry.SOURCE_TALK_TOPIC), null))
+        }
+    }
```

Comment:
If click on `User:username` link, the bottomsheet will show but the buttons and overflow menu items will not work.

Reply:
Done. (forgot to actually hook up the LinkPreviewDialog)